From c9c81c2b8a039268d2e0482c08ba63fa5af7b621 Mon Sep 17 00:00:00 2001
From: Johan Klokkhammer Helsing <johan.helsing@qt.io>
Date: Mon, 21 Jan 2019 11:56:03 +0100
Subject: [PATCH 07/26] Fix platform integration not building with
 -no-feature-wayland-server

The plugins subdir which includes the platform plugins was not built without
the feature wayland-server. Moves the checks inside the plugins folder to make
it build.

Also disables some tests that won't pass without the compositor API.

Fixes: QTBUG-73169
Change-Id: Ie68badd2ff62ee544c31edaff7c732082e0a4115
Reviewed-by: Paul Olav Tvete <paul.tvete@qt.io>
(cherry picked from commit f9ba56f24e7d1fcc9d0dc5eabb55335fc64b9a56)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 .../hardwareintegration/hardwareintegration.pro    |  2 +-
 src/src.pro                                        | 14 ++++++++------
 tests/auto/auto.pro                                |  6 +++++-
 3 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/src/plugins/hardwareintegration/hardwareintegration.pro b/src/plugins/hardwareintegration/hardwareintegration.pro
index 59d867f9..cb7a4b26 100644
--- a/src/plugins/hardwareintegration/hardwareintegration.pro
+++ b/src/plugins/hardwareintegration/hardwareintegration.pro
@@ -1,4 +1,4 @@
 TEMPLATE=subdirs
 
 SUBDIRS += client
-SUBDIRS += compositor
+qtHaveModule(waylandcompositor): SUBDIRS += compositor
diff --git a/src/src.pro b/src/src.pro
index 4ecbc71b..031ff885 100644
--- a/src/src.pro
+++ b/src/src.pro
@@ -13,6 +13,14 @@ qtConfig(wayland-client) {
     sub_client.target = sub-client
     SUBDIRS += sub_client
 
+    sub_plugins.subdir = plugins
+    sub_plugins.depends += sub-qtwaylandscanner sub-client
+    qtConfig(wayland-server) {
+        sub_plugins.depends += sub-compositor
+    }
+    sub_plugins.target = sub-plugins
+    SUBDIRS += sub_plugins
+
     qtConfig(wayland-server) {
         sub_compositor.subdir = compositor
         sub_compositor.depends = sub-qtwaylandscanner
@@ -23,11 +31,5 @@ qtConfig(wayland-client) {
         sub_imports.depends += sub-compositor
         sub_imports.target = sub-imports
         SUBDIRS += sub_imports
-
-        sub_plugins.subdir = plugins
-        sub_plugins.depends = sub-qtwaylandscanner sub-client sub-compositor
-        sub_plugins.target = sub-plugins
-        SUBDIRS += sub_plugins
     }
 }
-
diff --git a/tests/auto/auto.pro b/tests/auto/auto.pro
index 79ad29bd..4ae3b4f6 100644
--- a/tests/auto/auto.pro
+++ b/tests/auto/auto.pro
@@ -2,6 +2,10 @@ TEMPLATE=subdirs
 QT_FOR_CONFIG += waylandclient-private
 
 qtConfig(wayland-client): \
-    SUBDIRS += client cmake
+    SUBDIRS += client
+
+qtConfig(wayland-client):qtHaveModule(waylandcompositor): \
+    SUBDIRS += cmake
+
 qtHaveModule(waylandcompositor): \
     SUBDIRS += compositor
-- 
2.20.1

